### PR TITLE
Proposal for Issue on Issue-719

### PR DESCRIPTION
**Pull request by Hobbydos**
Proposal fix for [issue 719](https://github.com/openconfig/featureprofiles/issues/719)
cc @robshakir @xw-g

### CODE REVIEW

**Comment by xw-g** [comment]
cc: @jjing1122

**Comment by jjing1122** [comment]
During local run of route_propagation_test, i hit the following errors, it seems oc path is invalid. have you seen this error or the test passed in your local test environment?
"error receiving gNMI response: failed to render notifications: cannot represent field value [0xc0049bff60] as TypedValue for path /network-instances/network-instance[name=port2]/protocols/protocol[identifier=BGP][name=0]/bgp/rib/attr-sets/attr-set[index=2]/as-path/as-segment: invalid type ptr in leaflist"

based on the codes, it seems oc path should be something like :
/network-instances/network-instance[name=port2]/protocols/protocol[identifier=BGP][name=0]/bgp/rib/afi-safis/afi-safi[afi-safi-name=IPV4_UNICAST]/ipv4-unicast/neighbors/neighbor[neighbor-address:xxxx]/adj-rib-in-pre/routes/route[path-id=0][prefix=xxxx]/state/attr-index

**Comment by Hobbydos** [comment]
The error you mentioned is from `ixgnmi.go`:
```
E1115 12:10:51.229043   22905 ixgnmi.go:469] 
Failed to publish roots: 
failed to render notifications: cannot represent field value [0xc0026fefc0] as TypedValue for path:
/network-instances/network-instance[name=port2]/protocols/protocol[identifier=BGP][name=0]/bgp/rib/attr-sets/attr-set[index=0]/as-path/as-segment:
invalid type ptr in leaflist
```
which is generated by this block (just pasting `prefix.v4` for the sake of simplicity):
https://github.com/openconfig/featureprofiles/blob/45481794a66a41bb6dfea1bec98cdd4cd84d5dfc/feature/bgp/addpath/ate_tests/route_propagation_test/route_propagation_test.go#L310-L321

We only see this issue when testing our changes with yesterday's commit https://github.com/openconfig/featureprofiles/commit/d3d4ff7252906141715820ad0c48492f828c9a87. However, that gNMI path doesn't seem like it's returned from the DUT but returned by the ATE.

When this was tested a week ago, everything worked as expected.

**Comment by xw-g** [comment]
@jjing1122  The error mentioned is not related to this PR.  Any other concerns about this PR? LGTM? 

@vcheny @sachendras @vosipchu, `BGP-PEER-GROUP1` is referenced but not defined. This PR is to fix that. I think probably we only need 1 LGTM from one of you, just in case I missed anything before I merge it.

**Comment by xw-g** [comment]
@Hobbydos can you also help update the OTG part? Thanks.

**Comment by vosipchu** [comment]
Xiao, i see both groups, definitions and policies applied. LGTM

**Comment by jjing1122** [comment]
LGTM.

**Review by jjing1122 — DISMISSED** [review]

File: feature/bgp/addpath/ate_tests/route_propagation_test/route_propagation_test.go
Comment (vcheny) [inline feature/bgp/addpath/ate_tests/route_propagation_test/route_propagation_test.go:161]: Do we still need the group level afi-safi configured given that the same is configured on the global level? Consider removing group level ones... Same for the remaining.

**Review by vcheny — COMMENTED** [review]
Can we confirm the test passed w/ this PR?

File: feature/bgp/addpath/ate_tests/route_propagation_test/route_propagation_test.go
Comment (vcheny) [inline feature/bgp/addpath/ate_tests/route_propagation_test/route_propagation_test.go:167]: We need to configure PERMIT-ALL policy as well.

**Review by vcheny — COMMENTED** [review]
I would prefer ensuring the test passes before merging this request unless tests fail because of vendor issues.

File: feature/bgp/addpath/ate_tests/route_propagation_test/route_propagation_test.go
Comment (sachendras) [inline feature/bgp/addpath/ate_tests/route_propagation_test/route_propagation_test.go:156]: How are we checking for MRAI? (for more info https://datatracker.ietf.org/doc/html/rfc4271#section-9.2.1.1)

Following per the procedure for  RT-1.3: BGP Route Propagation

=================
Establish BGP sessions between:
ATE port-1 and DUT port-1
ATE port-2 and DUT port-2
For IPv4 and IPv6:
Advertise prefixes from ATE port-1, observe received prefixes at ATE port-2.
Specify default accept for received prefixes on DUT.
Specify table based neighbor configuration to cover - validating the supported capabilities from the DUT.
	- MRAI (minimum route advertisement interval), ensuring routes are advertised within specified time.  <<<<---
	- IPv4 routes with an IPv6 next-hop when negotiating RFC5549 - validating that routes are accepted and advertised with the specified values.
	- With ADD-PATH enabled, ensure that multiple routes are accepted from a neighbor when advertised with individual path IDs, and that these routes are advertised to ATE port-2.

Comment (Hobbydos) [reply feature/bgp/addpath/ate_tests/route_propagation_test/route_propagation_test.go]: @sachendras, Do you think this might be out of scope for this PR?

Comment (sachendras) [reply feature/bgp/addpath/ate_tests/route_propagation_test/route_propagation_test.go]: Sure

File: feature/bgp/addpath/ate_tests/route_propagation_test/route_propagation_test.go
Comment (jjing1122) [reply feature/bgp/addpath/ate_tests/route_propagation_test/route_propagation_test.go]: @Hobbydos Please confirm.

File: feature/bgp/addpath/ate_tests/route_propagation_test/route_propagation_test.go
Comment (jjing1122) [reply feature/bgp/addpath/ate_tests/route_propagation_test/route_propagation_test.go:186]: @Hobbydos please check.

Comment (Hobbydos) [reply feature/bgp/addpath/ate_tests/route_propagation_test/route_propagation_test.go:186]: those were added some time back in latest commit - See: https://github.com/openconfig/featureprofiles/blob/a39ddda34a80c6f697fb91e918f32a3c452ef78f/feature/bgp/addpath/ate_tests/route_propagation_test/route_propagation_test.go#L121-L126

File: feature/bgp/addpath/ate_tests/route_propagation_test/route_propagation_test.go
Comment (Hobbydos) [reply feature/bgp/addpath/ate_tests/route_propagation_test/route_propagation_test.go]: @jjing1122 cc @vcheny 
Peer group level afi-safi configs were removed in favor for global level ones;
commit removed c0cb515
in latest commit, you'll only see global and targeted neighbor configs such as 
https://github.com/openconfig/featureprofiles/blob/a39ddda34a80c6f697fb91e918f32a3c452ef78f/feature/bgp/addpath/ate_tests/route_propagation_test/route_propagation_test.go#L198-L207

**Review by jjing1122 — DISMISSED** [review]

**Comment by Hobbydos** [comment]
@jjing1122 conflicts should be handled.

**Review by jjing1122 — APPROVED** [review]

**Pull request merged**
